frontend/openchat-client/src/utils/list.ts

```diff
@@ -116,3 +116,16 @@ export function toRecordFiltered<T, K extends string | number | symbol, V>(
         return rec;
     }, {} as Record<K, V>);
 }
+
+export function keepMax<T>(items: T[], keep: (item: T) => boolean, max: number) {
```

Comment:
Makes me a bit queasy that this function is void and mutates the input.

Reply:
Does splice make you queasy in general?